tests/fib/test_fib.py

```diff
@@ -140,101 +145,89 @@ def test_fib(testbed, duthost, ptfhost, ipv4, ipv6, mtu):
                         "testbed_mtu": mtu,
                         "test_balancing": test_balancing },
                 log_file=log_file,
+                qlen=PTF_QLEN,
                 socket_recv_size=16384)
 
-@pytest.fixture(scope="module")
-def timestamp():
-    yield datetime.now().strftime('%Y-%m-%d-%H:%M:%S')
-
-@pytest.fixture(scope="function")
-def setup_hash(testbed, duthost, ptfhost, timestamp):
-    global g_vars
 
-    config_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
-
-    ofpname = "/tmp/fib/{}/tmp/fib_info.{}.txt".format(duthost.hostname, timestamp)
+@pytest.fixture(scope="module")
+def setup_hash(testbed, duthost, config_facts):
+    timestamp = datetime.now().strftime('%Y-%m-%d-%H:%M:%S')
 
-    build_fib(duthost, config_facts, ofpname, timestamp)
-
-    ptfhost.copy(src=ofpname, dest="/root/fib_info.txt")
-    logging.info("run ptf test")
+    setup_info = {}
 
     # TODO
-    if 'dst-mac' in HASH_KEYS:
-        HASH_KEYS.remove('dst-mac')
+    hash_keys = HASH_KEYS[:]    # Copy from global var to avoid side effects of multiple iterations
+    if 'dst-mac' in hash_keys:
+        hash_keys.remove('dst-mac')
 
     # do not test load balancing on L4 port on vs platform as kernel 4.9
     # can only do load balance base on L3
-    meta = config_facts.get('DEVICE_METADATA')
-    if meta['localhost']['platform'] == 'x86_64-kvm_x86_64-r0':
-        HASH_KEYS.remove('src-port')
-        HASH_KEYS.remove('dst-port')
-
-    g_vars['testbed_type'] = testbed['topo']['name']
-    g_vars['router_mac'] = duthost.shell('sonic-cfggen -d -v \'DEVICE_METADATA.localhost.mac\'')["stdout_lines"][0].decode("utf-8")
+    if duthost.facts['platform'] == 'x86_64-kvm_x86_64-r0':
```

Comment:
Same q as the other instance of platform.

Reply:
Updated.